test/parallel/test-internal-util-objects.js

```diff
@@ -29,6 +29,23 @@ Object.prototype.blep = 'blop';
     Object.getOwnPropertyNames(kEnumerableProperty),
     [ 'enumerable' ]
   );
+  assert.strictEqual(
+    Object.isExtensible(kEnumerableProperty),
+    false
+  );
+  assert.strictEqual(
+    Object.isSealed(kEnumerableProperty),
+    true
+  );
+  assert.strictEqual(
+    Object.isFrozen(kEnumerableProperty),
+    true
+  );
```

Comment:
This doesn't really seem like something we need to care in our tests (in no credible scenario will anyone tries to check what's the value of `Object.isSealed(kEnumerableProperty)` in core), the consequences are what's really important to test imo:
```suggestion
  assert.throws(
    () => { kEnumerableProperty.configurable = false; },
    TypeError
  );
  assert.throws(
    () => Object.assign(kEnumerableProperty, { configurable: false }),
    TypeError
  );
  assert.throws(
    () => Object.assign(kEnumerableProperty, { enumerable: false }),
    TypeError
  );
```

Reply:
Conceptually agreed. :)
My concern here is that by assuming alternatives to freezing and possibility of something dangerous (e.g. if there will be unavoidable exposing reference to `kEnumerableProperty` for userland) we must keep in mind every single way to break it.
If that approach will become common in core, we'll have to create an exhaustive `isImmutable()` helper function instead.